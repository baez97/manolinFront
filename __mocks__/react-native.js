function AsyncStorageClass() {
    this.setItem = jest.fn((key, value) => {
        return new Promise((resolve, reject) => {
            resolve();
        })
    });

    this.getItem = jest.fn((key, value) => {
        return new Promise((resolve, reject) => {
            resolve("THIS_IS_THE_SAVED_TOKEN");
        })
    });

    this.removeItem = jest.fn((key) => {
        return  new Promise((resolve, reject) => {
            resolve();
        })
    });

    this.removeItem.mockReturnValueOnce(new Promise((res,rej) => {
        rej({message: "THIS IS AN ERROR"});
    }))
}

module.exports.AsyncStorage = new AsyncStorageClass();