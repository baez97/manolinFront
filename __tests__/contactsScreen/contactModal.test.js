import React from 'react';
import { Linking } from 'react-native';
import renderer from 'react-test-renderer';
import ContactModal from '../../components/contactsComponents/contactModal';

jest.unmock("react-native")
jest.unmock('react')
jest.unmock("react-test-renderer")
jest.unmock('expo');

describe("ContactModal", () => {
    describe("View", () => {
        const mockedContact = {
            name : "CONTACT_NAME",
            phone: "CONTACT_PHONE" 
        };

        it("Matches the Snapshot when is not visible", () => {
            const view = <ContactModal 
                isVisible = { false }
                contact   = { mockedContact }/>;
            const rendered = renderer.create(view);
            const renderedJson = rendered.toJSON();

            expect( renderedJson ).toMatchSnapshot();
        });

        it("Matches the Snapshot when is not visible", () => {
            const view = <ContactModal 
                isVisible = { true} 
                contact   = { mockedContact }/>;
            const rendered = renderer.create(view);
            const renderedJson = rendered.toJSON();

            expect( renderedJson ).toMatchSnapshot();
        });
    });
});

