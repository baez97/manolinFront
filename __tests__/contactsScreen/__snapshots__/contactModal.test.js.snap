// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactModal View Matches the Snapshot when is not visible 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={false}
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 35.5,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "paddingBottom": 43.75,
            "paddingLeft": 26.25,
            "paddingRight": 26.25,
            "paddingTop": 26.25,
            "width": 602,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Object {
              "color": "#1b7128",
              "fontFamily": "montserrat-extra-bold",
              "fontSize": 40.7291678090925,
              "marginBottom": 17.5,
              "textAlign": "center",
            }
          }
        >
          CONTACT_NAME
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Object {
              "color": "#2db842",
              "fontFamily": "montserrat-extra-bold",
              "fontSize": 38.95833442608516,
              "textAlign": "left",
            }
          }
        >
          CONTACT_PHONE
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ExponentLinearGradient
            colors={
              Array [
                4279286482,
                4278481116,
              ]
            }
            endPoint={
              Array [
                0.65,
                1,
              ]
            }
            startPoint={
              Array [
                0.55,
                0,
              ]
            }
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 35.5,
                "elevation": 15,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginTop": 35,
                "paddingBottom": 17.5,
                "paddingLeft": 35,
                "paddingRight": 35,
                "paddingTop": 17.5,
                "width": 531,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Object {
                  "color": "white",
                  "fontFamily": "montserrat-extra-bold",
                  "fontSize": 38.95833442608516,
                  "textAlign": "center",
                }
              }
            >
              Llamar
            </Text>
            <Image
              resizeMode="contain"
              source={1}
              style={
                Object {
                  "height": 140,
                  "width": 140,
                }
              }
            />
          </ExponentLinearGradient>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ExponentLinearGradient
            colors={
              Array [
                4278439238,
                4278429498,
              ]
            }
            endPoint={
              Array [
                0.65,
                1,
              ]
            }
            startPoint={
              Array [
                0.55,
                0,
              ]
            }
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 35.5,
                "elevation": 15,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginTop": 35,
                "paddingBottom": 17.5,
                "paddingLeft": 35,
                "paddingRight": 35,
                "paddingTop": 17.5,
                "width": 531,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Object {
                  "color": "white",
                  "fontFamily": "montserrat-extra-bold",
                  "fontSize": 38.95833442608516,
                  "textAlign": "center",
                }
              }
            >
              WhatsApp
            </Text>
            <Image
              resizeMode="contain"
              source={1}
              style={
                Object {
                  "height": 140,
                  "width": 140,
                }
              }
            />
          </ExponentLinearGradient>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`ContactModal View Matches the Snapshot when is not visible 2`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 35.5,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "paddingBottom": 43.75,
            "paddingLeft": 26.25,
            "paddingRight": 26.25,
            "paddingTop": 26.25,
            "width": 602,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Object {
              "color": "#1b7128",
              "fontFamily": "montserrat-extra-bold",
              "fontSize": 40.7291678090925,
              "marginBottom": 17.5,
              "textAlign": "center",
            }
          }
        >
          CONTACT_NAME
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Object {
              "color": "#2db842",
              "fontFamily": "montserrat-extra-bold",
              "fontSize": 38.95833442608516,
              "textAlign": "left",
            }
          }
        >
          CONTACT_PHONE
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ExponentLinearGradient
            colors={
              Array [
                4279286482,
                4278481116,
              ]
            }
            endPoint={
              Array [
                0.65,
                1,
              ]
            }
            startPoint={
              Array [
                0.55,
                0,
              ]
            }
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 35.5,
                "elevation": 15,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginTop": 35,
                "paddingBottom": 17.5,
                "paddingLeft": 35,
                "paddingRight": 35,
                "paddingTop": 17.5,
                "width": 531,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Object {
                  "color": "white",
                  "fontFamily": "montserrat-extra-bold",
                  "fontSize": 38.95833442608516,
                  "textAlign": "center",
                }
              }
            >
              Llamar
            </Text>
            <Image
              resizeMode="contain"
              source={1}
              style={
                Object {
                  "height": 140,
                  "width": 140,
                }
              }
            />
          </ExponentLinearGradient>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ExponentLinearGradient
            colors={
              Array [
                4278439238,
                4278429498,
              ]
            }
            endPoint={
              Array [
                0.65,
                1,
              ]
            }
            startPoint={
              Array [
                0.55,
                0,
              ]
            }
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 35.5,
                "elevation": 15,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginTop": 35,
                "paddingBottom": 17.5,
                "paddingLeft": 35,
                "paddingRight": 35,
                "paddingTop": 17.5,
                "width": 531,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Object {
                  "color": "white",
                  "fontFamily": "montserrat-extra-bold",
                  "fontSize": 38.95833442608516,
                  "textAlign": "center",
                }
              }
            >
              WhatsApp
            </Text>
            <Image
              resizeMode="contain"
              source={1}
              style={
                Object {
                  "height": 140,
                  "width": 140,
                }
              }
            />
          </ExponentLinearGradient>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
